src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/SessionTracker.cs

```diff
@@ -0,0 +1,84 @@
+﻿using FluentResults;
+using System;
+using Tutor.Core.BuildingBlocks.EventSourcing;
+using Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries.Events;
+using Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries.Events.SessionLifecycleEvents;
+
+namespace Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries
+{
+    public class SessionTracker : EventSourcedEntity
+    {
+        public TimeSpan DurationOfPreviousSessions { get; private set; } = new TimeSpan(0);
+        public TimeSpan DurationOfCurrentSession { get => HasActiveSession ? LastActivity.Value - StartOfCurrentSession.Value : new TimeSpan(0); }
+        public bool HasActiveSession { get => StartOfCurrentSession.HasValue; }
+        public DateTime? StartOfCurrentSession { get; private set; }
+        public DateTime? LastActivity { get; private set; }
```

Comment:
Also, now that we're tracking the time of the last activity in a session, maybe we could add a simple improvement for session abandonment by implementing a method in the KCM aggregate which abandons the session if it's been inactive for a certain amount of time? Then we can add a scheduled process or something similar that would find all KCM's with active sessions, and abandon them if necessary.